packages/website/ts/pages/explore.tsx

```diff
@@ -0,0 +1,366 @@
+    public state = {
+        isTilesLoading: false,
+        isContactModalOpen: false,
+        tiles: [] as ExploreTile[],
+        tilesOrdering: ExploreTilesOrdering.None,
+        isEditorialShown: true,
+        filters: FILTERS,
+        query: '',
+    };
+
+    constructor(props: ExploreProps) {
+        super(props);
+    }
+
+    public componentWillMount(): void {
+        // tslint:disable-next-line:no-floating-promises
+        this._loadEntriesAsync().then(() => {
+            this._setFilter('all');
+        });
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <SiteWrap theme="light">
+                <DocumentTitle {...documentConstants.EXPLORE} />
+                <ExploreHero onSearch={this._changeSearchResults} />
+                <Section padding={'0 0 120px 0'} maxWidth={'1150px'}>
+                    <ExploreToolBar
+                        onFilterClick={this._setFilter}
+                        filters={this.state.filters}
+                        editorial={this.state.isEditorialShown}
+                        onEditorial={this._onEditorial}
+                        orderings={ORDERINGS}
+                        activeOrdering={this.state.tilesOrdering}
+                        onOrdering={this._onOrdering}
+                    />
+                    <ExploreGrid tiles={this._generateTilesFromState()} />
+                </Section>
+                <Banner
+                    heading="Working on a 0x project?"
+                    subline="Lorem Ipsum something then that and say something more."
+                    mainCta={{ text: 'Apply Now', onClick: this._onOpenContactModal }}
+                    secondaryCta={{ text: 'Join Discord', href: 'https://discordapp.com/invite/d3FTX3M' }}
+                />
+                <ModalContact
+                    isOpen={this.state.isContactModalOpen}
+                    onDismiss={this._onDismissContactModal}
+                    modalContactType={ModalContactType.Explore}
+                />
+            </SiteWrap>
+        );
+    }
+
+    private readonly _onOpenContactModal = (): void => {
+        this.setState({ isContactModalOpen: true });
+    };
+
+    private readonly _onDismissContactModal = (): void => {
+        this.setState({ isContactModalOpen: false });
+    };
+
+    private readonly _onEditorial = (newValue: boolean): void => {
+        // tslint:disable-next-line:no-floating-promises
+        this._generateTilesWithModifier(this.state.tiles, ExploreTilesModifiers.Editorial, {
+            isEditorialShown: newValue,
+        }).then(newTiles => {
+            this.setState({ isEditorialShown: newValue, tiles: newTiles });
+        });
+    };
+
+    private readonly _onOrdering = (newValue: string) => {
+        this.setState({ tilesOrdering: newValue });
+        // tslint:disable-next-line:no-floating-promises
+        this._generateTilesWithModifier(this.state.tiles, ExploreTilesModifiers.Ordering, {
```

Comment:
we use the `await` syntax instead of `then`.

So this would be 
```
const newTiles = await this._generateTilesWithModifier(..);
this.setState(...)
```